src/ClientData/CallstackDataTest.cpp

```diff
@@ -200,4 +200,124 @@ TEST(CallstackData, FilterCallstackEventsBasedOnMajorityStartExcludesFunctionToS
                                                                          event4, event5, event6}));
 }
 
+TEST(CallstackData, ForEachCallstackEventOfTidInTimeRangeDiscretized) {
```

Comment:
I suggest to split these tests into small focused cases (see http://go/tott/520)